packages/ffe-buttons-react/src/ExpandButton.js

```diff
@@ -1,37 +1,49 @@
 import React from 'react';
-import { bool, func, node } from 'prop-types';
+import { bool, func, oneOfType, string, node } from 'prop-types';
 import classNames from 'classnames';
+import KryssIkon from '@sb1/ffe-icons-react/lib/kryss-ikon';
+import Button from './BaseButton';
 
-export default function ExpandButton(props) {
-    const { children, isExpanded, onClick, ...rest } = props;
-
+const ExpandButton = (props) => {
+    const {
+        children,
+        className,
+        isExpanded,
+        ...rest
+    } = props;
     return (
-        <button
-            aria-expanded={String(isExpanded)}
-            className={classNames('ffe-expand-button', {
-                'ffe-expand-button--expanded': isExpanded,
-            })}
-            onClick={onClick}
+        <Button 
+            aria-expanded={isExpanded}
+            buttonType="expand" 
+            className={classNames(
+                { 'ffe-button--expanded': isExpanded },
+                className,
+            )}
+            leftIcon={isExpanded && <KryssIkon />}
```

Comment:
If I provide a `leftIcon` I would expect it to apply only to the label in its unexpanded state. Now it will override the `KryssIkon` when `isExpanded` is `true`. `KryssIcon` should perhaps instead be rendered instead of `children`?

That being said, `ExpandButton` should probably not build on `BaseButton` here in React IMO. We don't have an `isLoading` state, and so rendering the spinner and its wrapper seems a bit pointless.

Reply:
`InlineBaseButton` would be a better parent I think :) Only, this isn't necessarily inline. Hmm..

Reply:
Some good points! 
The reason why this abstraction was used is because the `ffe-button--expand` extends the look and feel of the `ffe-button--secondary`. It shares the same common less-class as well (`ffe-button`), and it follows a lot of the same usage patterns.

As you might have noticed, I added more full-featured prop types to each of the components than they had previously. This is to specify the supported API better than what the previous implementation did. I considered destructuring waay unsupported properties so that misuse would be even harder, but decided against it. I believe undocumented usage should be discouraged, not defensively coded against.

That being said, I'll try to implement a custom component that tries to solve this a bit better.